runtime/capsules/runtime/src/mctp/mctp_binding.rs

```diff
@@ -0,0 +1,204 @@
+// Licensed under the Apache-2.0 license
+
+use i3c_driver::hil::{I3CTarget, RxClient, TxClient};
+
+use core::cell::Cell;
+
+use kernel::debug;
+use kernel::utilities::cells::OptionalCell;
+use kernel::utilities::cells::TakeCell;
+use kernel::ErrorCode;
+
+// TODO: Set the correct value for MCTP_I3C_MAXBUF.
+pub const MCTP_I3C_MAXBUF: usize = 69; // 4 MCTP header + 64 baseline payload + 1 (PEC)
+
+pub const MCTP_I3C_MAXMTU: usize = MCTP_I3C_MAXBUF - 1; // 64 bytes
+pub const MCTP_I3C_MINMTU: usize = 4 + 64; // 4 MCPT header + 64 baseline payload
+
+/// This trait contains the interface definition
+/// for sending the MCTP packet through MCTP transport binding layer.
+pub trait MCTPTransportBinding<'a> {
+    /// Set the client that will be called when the packet is transmitted.
+    fn set_tx_client(&self, client: &'a dyn TxClient);
```

Comment:
I wonder if we should use our own separate `TxClient` and `RxClient` interfaces for MCTP that are different from the I3C ones.

Specifically, the I3C ones have the weird thing where reads and writes are backwards (since for I3C, we transmit read responses and receive writes).

It also helps keeps the layers a little more independent. (In software engineering terms, we call the layers blending together too much to be a "leaky abstraction".)

Reply:
Although they look similar, I introduced new traits for the transport binding layer to distinguish them from those in HIL.